pkg/cloudinfo/cloudinfo.go

```diff
@@ -251,14 +262,21 @@ func (cpi *CachingCloudInfo) renewProviderInfo(ctx context.Context, provider str
 	}
 	log.Info("finished to renew products (vm-s)")
 
-	if _, err := cpi.renewStatus(provider); err != nil {
+	// close the span
+	cpi.tracer.EndSpan(vmCtx)
```

Comment:
This should be called even if the function returns earlier with an error. Errors should also be somehow present in the trace. Check ochttp for an example.

Reply:
yes, this is only a dirty solution for thenew span - probably the code needs to be adjusted so that the span can be made a method call

Reply:
raised issue about proper span delimitation / this can be either removed or left and commented here